walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,135 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
+  /*
+   * Assigns attendees from request to local HashMap to search for them in constant time.
+   */
+  private HashSet<String> toSet(MeetingRequest request) {
+    HashSet<String> requestAttendees = new HashSet<String>();
+    for(String attendee : request.getAttendees()) {
+      requestAttendees.add(attendee);
+    }
+    return requestAttendees;
+  }
+
+  /*
+   * Adds a TimeRange depending on the scenario. Some additions use a recursive call to this same
+   * function because if a new TimeRange is formed from a merge or an existing TimeRange is replaced
+   * because of a containment issue, it needs to be checked against all the previously added
+   * TimeRanges for new overlap or containment issues.
+   */
+  private ArrayList<TimeRange> addTimeRange(ArrayList<TimeRange> occupiedTimeSlots,
```

Comment:
It would be a good exercise to add to your algorithmic methods what the big-O analysis is both in terms of computation and memory! :) Let me know if you need any clarification on that!

Reply:
This would be included in the JavaDoc right?

Reply:
Yep exactly!

Reply:
Friendly reminder to write some big-O analysis! Otherwise everything looks good to me!